### PR TITLE
Issue-15: Add iOS sample

### PR DESCRIPTION
**Pull request by tobiasheine**
### Description
This adds a swift example of how to use the library.

This addresses https://github.com/novoda/aws-v4-signer-java/issues/15.

### Considerations

- add a swift friendly constructor of the `Builder` as swift doesn't know the concept of default params
- add end-to-end unit test
- add kotlin & swift samples to readme

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,8 +1,8 @@
-# aws-v4-signer-java
+# aws-v4-signer
```
Comment (tobiasheine) [inline README.md:1]: This readme needs more work and will be changed once we do the release and merge everuthing back to master.

**Review by florianmski — APPROVED** [review]
Looking legit 👍

**Review by juankysoriano — APPROVED** [review]

**Comment by juankysoriano** [comment]
LGMT

**Comment by zegnus** [comment]
wohoooo 💯

**Pull request merged**